docs/jax-101/05.1-pytrees.md

```diff
@@ -187,6 +187,22 @@ plt.scatter(xs, forward(params, xs), label='Model prediction')
 plt.legend();
 ```
 
+## Key paths
+
+In a nested pytree, each leaf has a unique "path of keys" corresponding to its place in the tree. JAX allows you to access the key paths and use them when handling a pytree. The corresponding APIs are:
+
+* `jax.tree_util.tree_flatten_with_path`
+* `jax.tree_util.tree_map_with_path`
+
+One use case is to print debugging information related to a certain leaf value:
+
+```{code-cell} ipython3
+tree = [1, {'k1': 2, 'k2': (3, 4)}, 5]
+flattened, _ = jax.tree_util.tree_flatten_with_path(tree)
+for key_path, value in flattened:
+    print(f'Value of tree{jax.tree_util.keystr(key_path)}: {value}')
+```
+
```

Comment:
Are `SequenceKey`, `DictKey`, etc in the public API? They're imported in `jax/tree_util.py`, but I forget what we decided.

If they're in the public API then we should document them. The conservative thing would be not to document them, but then should we remove them from `jax/tree_util.py`?

Reply:
Added with a showcase paragraph.